app/controllers/users_controller.rb

```diff
@@ -1,29 +1,30 @@
 class UsersController < ApplicationController
-  LOCALES = [
-              ["English", "en"],
-            ]
+  before_filter :load_resource
 
-  def settings
-    load_locales
+  def edit
+    return unless authorize_resource(@user, READ_USER)
   end
 
-  def update_settings
-    params = user_params
-    if current_user.update(params)
-      redirect_to settings_path, notice: "Settings updated"
-    else
-      load_locales
-      render :settings
+  def update
+    return unless authorize_resource(@user, UPDATE_USER)
+    if can_edit? && @user.update(user_params)
+      @_message = 'User updated'
     end
```

Comment:
We should split these cases:
- If cannot edit, then `head :json` and return, like in [authorize_resource](https://github.com/instedd/cdx/blob/feature/568-map_site/app/controllers/application_controller.rb#L44)
- If cannot update, due to a validation error, then `render edit` with the current model
- Only if it was successful redirect to `edit_user_path`

Note that with these changes there is no need for `@_message` to be an instance var